src/containers/Application/ApplicationPageWrapper.tsx

```diff
@@ -111,4 +164,83 @@ function processRedirect(appState: any): void {
   }
 }
 
+interface buildProgressInApplicationProps {
+  elementsState: ApplicationElementStates | undefined
+  responses: ResponsesFullByCode | undefined
+  templateSections: TemplateSectionPayload[]
+  isLinear: boolean
+  currentSection: number
+  currentPage: number
+}
+
+function buildProgressInApplication({
+  elementsState,
+  responses,
+  templateSections,
+  isLinear,
+  currentSection,
+  currentPage,
+}: buildProgressInApplicationProps): ProgressInApplication {
+  if (!elementsState || !responses) return []
+  let sectionsStructure: ProgressInApplication = templateSections.map((section) => {
+    // Create an array with all pages in each section
+    const pages = Array.from(Array(section.totalPages).keys(), (n) => n + 1)
+    const checkIsCurrentPage = (page: number) =>
+      section.index === currentSection && page === currentPage
+    const checkIsPreviousPage = (index: number, page: number) =>
+      index < currentSection ? true : index === currentSection && page <= currentPage
+
+    // Build object to keep each section progress (and pages progress)
+    const progressInSection: ProgressInSection = {
+      code: section.code,
+      title: section.title,
+      canNavigate: isLinear ? section.index <= currentSection : true,
+      isActive: section.index === currentSection,
+      pages: pages.map((number) => ({
+        pageName: `Page ${number}`,
+        canNavigate: isLinear ? checkIsPreviousPage(section.index, number) : true,
+        isActive: checkIsCurrentPage(number),
+        status: checkIsCurrentPage(number)
```

Comment:
can you comment, what this logic is here, shouldn't it just be :`validatePage(...)` ?

I would do pages outside of the progressInSection object like this:

```

const pages = ...

progressInSection: {
..
status: getCombinedStatus(pages),
pages
}

```

Reply:
Moved to outside of the scope as suggested: [here](https://github.com/openmsupply/application-manager-web-app/pull/127/commits/b6e437b958e873fd0182e53fa8fa4174cc4a91c9)
 
The reason why the `validatePage` is done 2 ways is differing if it would run in a `strict` or `loose` mode.
1. `linear` and `previous page` it would run a STRICT validation
2. `non-linear` and `previous page` is should run a LOOSE validation

Now in case the user has been to the `Summary` page I think we would need to run a STRICT validation over all the application, which is not implemented yet. I think it would be better if we leave this to be dealt in the issue #48